### DISCUSSION TITLE
Support tuple syntax on subqueries for SQL compatibility

### DISCUSSION THREAD

**Discussion by takacsot** [duckdb/duckdb#14472] [Ideas]
The following queries are all working fine on Postgresql and other databases:

```sql
select 1 where (1,2)  = (1,2);
select 1 where (1,2) in ((1,2), (3,4));
select 1 where (1,2) in (select 1,2);
```

In DuckDb only the first 2 what is working. 

The last one is throwing error of "Binder Error: Subquery returns 2 columns - expected 1 LINE 1: select 1 where (1,2) in (select 1,2)"

The second statement is proving that there are some level of support but the third one is not supported.

**Comment by soerenwolfers** [comment]
just write `select 1 where (1,2) in (select (1,2));`

**Reply by soerenwolfers** [reply]
given that your original attempt works with PostgreSQL (as odd as it is) and duckdb typically strives for compatibility with PostgreSQL you have a chance the current behavior will be considered a big and fixed if you report a regular issue (rather than a discussion). Nonetheless, I think adding the parentheses makes it cleaner.

**Reply by takacsot** [reply]
Thank you very much. Indeed it helps :)

**Reply by takacsot** [reply]
But I will open as bug as the suggested solution is not working on Postgresql :( so If postgresql compatibility is a hard goal this needs to improve.
